### PR TITLE
Goliath-Infused Tendril Hammer uses an internal cooldown for the its special attack instead of a universal click cooldown

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Rather than using a click cooldown, the tendril hammer instead can make its special heavy attack every 2 seconds.

## Why It's Good For The Game

In my newfound quest to try and eliminate universal click cooldowns or weird non-interactivity timers as balancing factors, this definitely is one of the biggest standout offenders. Lemme make an argument for universal click cooldowns increases being an ineffective limitation. 

I'll use the problems presented by the tendril hammer to highlight some of those problems, as well as unique problems to the tendril hammer itself.

<details>
  <summary>da big discussion</summary>

A) The functionality of the hammer actively inhibits all in-game handuse interaction for several seconds, without explaining this to a player. As a player, you won't know why this is happening, as universal click cooldown is not present as a UI element.

B) Since universal click cooldowns are not visible to players, it might feel more like the game is malfunctioning rather than being a deliberate mechanic. Even if click cooldowns were visible, players probably would think that the cooldown applies to the hammer, and not handuse interactivity with the game world as a whole for several seconds.

C) The functionality of the hammer could work fine as an internal cooldown on the hammer, only relevant to the hammer. This ensures that its special effects are exclusive, without the need to interrupt player interaction as a whole.

D) Since we're talking about miners. If someone is concerned about the hammer being used on the station against carbon players; you need someone to help mutate you into goliath mutant, which cannot be bypassed whatsoever. An excellent example of something similar is the chainsaw arm, created right next door to genetics in robotics, which does even more force than the arm and is sharp. With the limitations that exist, I think it probably discourages most powergaming, if that was even a realistic concern (it really isn't).

E) You lose both a hand AND your gloves slot when you get the hammer. No modsuits, no glove equipment, no two-handed equipment, and you now have to juggle everything with one hand assuming you're not on your, once again, universal click cooldown for several precious seconds. Miners live or die in their rapid response to problems. This is also the total sum of what you lose as a miner. That's a steep cost and it just doesn't justify its own value compared to what you lose.

</details>

TL;DR - There is no offset to the cost of this weapon, it is strictly a detriment because of poorly conceived implementation.

This is maybe one of the coolest ideas conceptually for the infusions so far, heavily hampered by what seems to be an intense fear of the mutation being _too useful_. So it was made _borderline masochistic to willingly seek out and use_.

I want to see this actually be useful. I can't see this with the restrictions it has. Hopefully this is enough to make it worthwhile getting.

## Changelog
:cl:
balance: Changes the universal click cooldown of the tendril hammer from the goliath infusion into an internal cooldown just for the special heavy attack.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
More or less.

**Comment by Jacquerel** [comment]
We could also do with some sort of visual indicator for _secondary_ cooldowns like this but that's more musing than a requirement for this PR.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 1]
```diff
@@ -113,29 +113,44 @@
 	toolspeed = 0.1
 	/// Amount of damage we deal to the mining and boss factions.
 	var/mining_bonus_force = 80
+	COOLDOWN_DECLARE(tendril_hammer_cd)
```
Comment (Jacquerel) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:116]: You should doc comment these the same way as var declarations (given that it is one. with a weird wrapper)

**Comment by necromanceranne** [comment]
> What is the reasoning behind moving it to secondary attack? 

While I do think you would want to use it as often as possible, having some control over when it occurs might just be stronger player agency. Maybe they want to save it, maybe they want to use it at any given time. Having that option open is I think a positive without much cost, unless you think it would actively make the arm less intuitive.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm (hunk 1 above)
Comment (necromanceranne) [reply code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:116]: Sure

**Comment by Jacquerel** [comment]
> While I do think you would want to use it as often as possible, having some control over when it occurs might just be stronger player agency. Maybe they want to save it, maybe they want to use it at any given time. Having that option open is I think a positive without much cost, unless you think it would actively make the arm less intuitive.

I do think it is less intuitive a little, especially when the intent here is to remove invisible timers.
Moving it to a right click reduces the _impact_ of the invisible timer, but still means that you're not quite sure when you should be right or left clicking. 
There almost essentially isn't any time when this would be available for use and you _wouldn't_ immediately want to use it.

**Comment by necromanceranne** [comment]
I can swap those back, no problem.

**Comment by catrock31** [comment]
could you add a sprite of the arm glowing slightly in your inventory to show when the super punch is available

File: code/game/machinery/dna_infuser/infuser_entries/infuser_tier_one_entries.dm [hunk 2]
```diff
@@ -10,10 +10,10 @@
 	desc = "The guy who said 'Whoever fights monsters should see to it that in the process he does not become a monster' clearly didn't see what a goliath miner can do!"
 	threshold_desc = "you can walk on lava!"
 	qualities = list(
-		"can breath both the station and lavaland air but can't deal with pure O2",
+		"can breath both the station and lavaland air, but be careful around pure 02",
```
Comment (Jacquerel) [inline code/game/machinery/dna_infuser/infuser_entries/infuser_tier_one_entries.dm:13]:
```suggestion
		"can breath both the station and lavaland air, but be careful around pure oxygen",
```
Wait did you replace this O with a 0?
Frankly it should probably just say the gas name not its atomic composition.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm [hunk 3]
```diff
@@ -113,29 +113,43 @@
...
 
 /obj/item/goliath_infuser_hammer/Initialize(mapload)
 	. = ..()
 	ADD_TRAIT(src, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
 
-/obj/item/goliath_infuser_hammer/melee_attack_chain(mob/user, atom/target, params)
+/obj/item/goliath_infuser_hammer/examine(mob/user)
 	. = ..()
-	user.changeNext_move(CLICK_CD_MELEE * 2) //hits slower but HARD
+	. += "You can use your tendril hammer arm to deliver a devastating blow against mining fauna by <b>right-clicking</b> them with the arm."
```
Comment (Jacquerel) [inline code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:125]:
```suggestion
	. += "You can use your tendril hammer arm to deliver a devastating blow against mining fauna, but only once every two seconds."
```
Existing description isn't true any more.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Will merge once these are fixed.

File: code/game/machinery/dna_infuser/infuser_entries/infuser_tier_one_entries.dm (hunk 2 above)
Comment (necromanceranne) [reply code/game/machinery/dna_infuser/infuser_entries/infuser_tier_one_entries.dm:13]: What the fuck, uh, my b

Yeah sure no worries I agree with that.

File: code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm (hunk 3 above)
Comment (necromanceranne) [reply code/game/machinery/dna_infuser/organ_sets/goliath_organs.dm:125]: knew I forgot something
